packages/flutter/test/widgets/container_test.dart

```diff
@@ -498,6 +498,51 @@ void main() {
       ),
     );
   });
+
+  testWidgets('giving clipBehaviour null or Clip.None, will not add a ClipPath to the tree', (WidgetTester tester) async {
+    await tester.pumpWidget(Container(
+      clipBehavior: Clip.none,
+      decoration: BoxDecoration(
+        borderRadius: BorderRadius.circular(1),
+      ),
+      child: const SizedBox(),
+    ));
+
+    expect(
+      find.byType(ClipPath),
+      findsNothing,
+    );
+
+    await tester.pumpWidget(Container(
+      clipBehavior: null,
+      decoration: BoxDecoration(
+        borderRadius: BorderRadius.circular(1),
+      ),
+      child: const SizedBox(),
+    ));
+
+    expect(
+      find.byType(ClipPath),
+      findsNothing,
+    );
+  });
+
+  testWidgets('giving clipBehaviour not a null, will add a ClipPath to the tree', (WidgetTester tester) async {
+    final Container container = Container(
+      clipBehavior: Clip.hardEdge,
+      decoration: BoxDecoration(
+        borderRadius: BorderRadius.circular(1),
+      ),
+      child: const SizedBox(),
+    );
+
+    await tester.pumpWidget(container);
+
+    expect(
+      find.byType(ClipPath),
+      findsOneWidget,
+    );
+  });
```

Comment:
There should also be additional tests verifying that ShapeBorder.getClipPath and BoxBorder.getClipPath return the right thing.